Cdn_TotalCdn_Auto_Configure.php

```diff
@@ -476,14 +486,20 @@ public static function admin_notices() {
 	 *
 	 * @since x.x.x
 	 *
-	 * @param bool   $cdn_enabled Whether the CDN is enabled.
-	 * @param string $cdn_engine  The CDN engine.
-	 * @param string $api_key     The API key.
-	 * @param string $cdn_status The CDN status.
-	 *
+	 * @param  bool   $cdn_enabled Whether the CDN is enabled.
+	 * @param  string $cdn_engine  The CDN engine.
+	 * @param  string $api_key     The API key.
+	 * @param  string $cdn_status The CDN status.
 	 * @return bool True if the notice was shown, false otherwise.
 	 */
-	public static function maybe_show_auto_config_notice( $cdn_enabled, $cdn_engine, $api_key, $cdn_status ) {
+	public static function maybe_show_auto_config_notice( $cdn_enabled, $cdn_engine, $api_key, $cdn_status ): bool {
+		// Display notice only if on a W3TC page.
+		$page = \filter_input( INPUT_GET, 'page', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+
+		if ( empty( $page ) || strpos( $page, 'w3tc_' ) !== 0 ) {
+			return false;
+		}
+
 		// If the CDN is enabled and the engine is set, do not show the notice.
 		if ( $cdn_enabled && 'totalcdn' === $cdn_engine ) {
```

Comment:
There's a bug with this I've noticed where if the checkout is successful and the auto-configure successfully applies the settings but fails to generate a pull zone it shows the notice despite the auto-configure failing. During initial testing my licensing was pointed at staging CDNAPI but my website was pointed at my dev CDNAPI so the request to generate a pull zone failed

Maybe add a check to see if the pullzone exists and is authenticated, or something?